pkg/cloudprovider/servicecontroller/servicecontroller.go

```diff
@@ -465,3 +510,72 @@ func hostsFromNodeList(list *api.NodeList) []string {
 	}
 	return result
 }
+
+// watchNodes handles updating the hosts pointed to by all external load
+// balancers whenever the set of nodes in the cluster changes.
+func (s *ServiceController) watchNodes(nodeLister *cache.StoreToNodeLister, period time.Duration) {
```

Comment:
That code doesn't actually get notified when there's a change -- the ListWatch simply tells the Reflector object how to list and watch the objects it's interested in. The Reflector then updates the Store whenever it receives an update, but doesn't notify the person that created these objects.

There is an UndeltaStore which notifies the creator when a change has come in while still allowing full list of all the objects. However, these lists that we're doing here aren't actually polling the apiserver, they're polling the local cache held in the store. The current approach has the benefit that we do retries pretty much automatically, at the cost of just a little CPU to examine the local cache every once in a while. It's also closest to how the old version of the code worked.

I can change it to trigger reconciliation on UndeltaStore updates if you'd like, but I don't think it'll buy much, and it'll add complexity to get retries working.